### ISSUE TITLE
user passed in validate overwrites `limit: false`

### ISSUE THREAD

**Issue by stephen-willoughby** [express-rate-limit/express-slow-down#46]
### Description

I get lots of warnings like:

```
ChangeWarning: Setting limit or max to 0 disables rate limiting in express-rate-limit v6 and older, but will cause all requests to be blocked in v7 See https://express-rate-limit.github.io/WRN_ERL_MAX_ZERO/ for more information.
```

looking at the code I can see you try to take care of this: https://github.com/express-rate-limit/express-slow-down/blob/eb8734c7e84e6f94d88ab45a6f3d430bbe16450e/source/slow-down.ts#L143

But I'm passing in my own `validate` to suppress `delayMs` warnings.  This is then inside `notUndefinedOptions` and so its spreading overrides your addition on line 143

### Library version

2.0.0

### Node version

v18.17.1

### Module system

CommonJS

**Comment by nfriedly** [comment]
Oof, yeah, I think 

https://github.com/express-rate-limit/express-slow-down/blob/eb8734c7e84e6f94d88ab45a6f3d430bbe16450e/source/slow-down.ts#L147-L148

should come before 

https://github.com/express-rate-limit/express-slow-down/blob/eb8734c7e84e6f94d88ab45a6f3d430bbe16450e/source/slow-down.ts#L142-L145

**Comment by nfriedly** [comment]
Hum, I went to add a test, but there already are tests that should have caught this:

https://github.com/express-rate-limit/express-slow-down/blob/eb8734c7e84e6f94d88ab45a6f3d430bbe16450e/test/library/options-test.ts#L38-L53

So now I want to know why they're not working...

**Comment by nfriedly** [comment]
Oh, I think it's a timing issue: the test is synchronous, but the warning is logged asynchronously...

**Comment by nfriedly** [comment]
Ok, the fix should be released in v2.0.1 shortly
